test/core/telemetry/metrics_test.cc

```diff
@@ -629,6 +629,19 @@ TEST_F(MetricsTest, DisableByDefaultMetricIsNotRecordedByFakeStatsPlugin) {
             absl::nullopt);
 }
 
+TEST_F(MetricsTest, FindInstrumentByName) {
+  auto uint64_counter_handle =
+      GlobalInstrumentsRegistry::RegisterUInt64Counter(
+          "uint64_counter", "A simple uint64 counter.", "unit", true)
+          .Labels("label_key_1", "label_key_2")
+          .OptionalLabels("optional_label_key_1", "optional_label_key_2")
+          .Build();
+  auto instrument =
+      GlobalInstrumentsRegistry::FindInstrumentByName("uint64_counter");
+  EXPECT_TRUE(instrument.has_value());
+  EXPECT_EQ(instrument.value().index, uint64_counter_handle.index);
```

Comment:
Might be able to write these 2 EXPECTs with one:

EXPECT_THAT(instrument, ::testing::Optional(::testing::Field(&GlobalInstrumentsRegistry::GlobalInstrumentHandle::index, ::testing::Eq(uint64_counter_handle.index))));

Your call though if you want to change it or not :).

Reply:
Done, thanks for the suggestion.